core/transfer/registry/registry.go

```diff
@@ -43,35 +44,30 @@ func init() {
 }
 
 // NewOCIRegistry initializes with hosts, authorizer callback, and headers
-func NewOCIRegistry(ref string, headers http.Header, creds CredentialHelper) *OCIRegistry {
```

Comment:
I wonder if we should use the options pattern here. If we are going to stabilize this interface in 2.0 and we still have other "maybe implement" features in the type, then it would be good to have a way to extend the functionality further without adding a new functions.

Reply:
Thanks for the comment. Fixed it to use options.